src/RpcServer/RpcServer.Wallet.cs

```diff
@@ -1,187 +1,92 @@
-                    }
-                case "sendmany":
-                    {
-                        int to_start = 0;
-                        UInt160 from = null;
-                        if (_params[0] is JString)
-                        {
-                            from = _params[0].AsString().ToScriptHash();
-                            to_start = 1;
-                        }
-                        JArray to = (JArray)_params[to_start];
-                        return SendMany(from, to);
-                    }
-                case "sendtoaddress":
-                    {
-                        UInt160 assetId = UInt160.Parse(_params[0].AsString());
-                        UInt160 scriptHash = _params[1].AsString().ToScriptHash();
-                        string value = _params[2].AsString();
-                        return SendToAddress(assetId, scriptHash, value);
-                    }
-                default:
-                    return null;
-            }
-        }
-
-        public void PostProcess(HttpContext context, string method, JArray _params, JObject result)
-        {
-            switch (method)
-            {
-                case "invoke":
-                case "invokefunction":
-                case "invokescript":
-                    ProcessInvoke(result);
-                    break;
-            }
-        }
-
-        private void ProcessInvoke(JObject result)
-        {
-            if (Wallet != null)
-            {
-                Transaction tx = Wallet.MakeTransaction(result["script"].AsString().HexToBytes());
-                ContractParametersContext context = new ContractParametersContext(tx);
-                Wallet.Sign(context);
-                if (context.Completed)
-                    tx.Witnesses = context.GetWitnesses();
-                else
-                    tx = null;
-                result["tx"] = tx?.ToArray().ToHexString();
-            }
-        }
+        private Wallet wallet;
 
         private void CheckWallet()
         {
-            if (Wallet is null)
+            if (wallet is null)
                 throw new RpcException(-400, "Access denied");
         }
 
-        private JObject SignAndRelay(Transaction tx)
+        [RpcMethod]
+        private JObject CloseWallet(JArray _params)
         {
-            ContractParametersContext context = new ContractParametersContext(tx);
-            Wallet.Sign(context);
-            if (context.Completed)
-            {
-                tx.Witnesses = context.GetWitnesses();
-                System.LocalNode.Tell(new LocalNode.Relay { Inventory = tx });
-                return tx.ToJson();
-            }
-            else
-            {
-                return context.ToJson();
-            }
+            wallet = null;
+            return true;
         }
 
-        private JObject DumpPrivKey(UInt160 scriptHash)
+        [RpcMethod]
+        private JObject DumpPrivKey(JArray _params)
```

Comment:
We should have another Attribute for `RequireAuth` in order to call wallet methods.

Reply:
Currently if you have set `RpcUser` and `RpcPass` in config.json, you need to authenticate for any method.

Reply:
Maybe we should disallow some methods by default in the config. And if the user want to use it, they can allow them and set a password

Reply:
https://github.com/neo-project/neo-modules/blob/fae8c17baffde1cf105bd21f8eed35e9a0f96eda/src/RpcServer/RpcServer/config.json#L12

`openwallet` is disabled by default.

Reply:
But if you install the rpc, and open a wallet by your self, you will have troubles. We should disable by default all the wallet methods

Reply:
Hey Shargon, due to my tests，all wallet methods are disabled by default unless you invoke "openwallet" methods，it's absolutely not related to you open your wallet or not in local, they are different business.

Reply:
https://github.com/neo-project/neo-modules/pull/160#discussion_r356434877

Reply:
Then is good to me, thanks for clarify!